packages/shylu/shylu_dd/bddc/src/shylu_KrylovSolver.h

```diff
@@ -69,42 +70,21 @@ template <class SX, class SM, class LO, class GO>
   }
 
   KrylovSolver
-    (RCP< PreconditionerBDDC<SX,SM,LO,GO> > Preconditioner,
+    (PreconditionerBase<SX,SM,LO,GO>* Preconditioner,
+     Teuchos::ParameterList Parameters)
+  {
+    m_Preconditioner = rcp(Preconditioner);
+    m_Parameters = rcp( new Teuchos::ParameterList(Parameters) );
+    initialize();
+  }
+
+  KrylovSolver
+    (RCP< PreconditionerBase<SX,SM,LO,GO> > Preconditioner,
      RCP<Teuchos::ParameterList> Parameters) :
   m_Preconditioner(Preconditioner),
-    m_Parameters(Parameters),
-    m_interfacePreconditioner
-    (Parameters->get("Interface Preconditioner", false)),
-    m_calculateRelativeResidualActual(false),
-    m_vectorLength(Preconditioner->NumMyRows()),
-    m_krylovLength(Preconditioner->NumMyRowsKrylov()),
-    m_maxIter
-    (Parameters->get("Maximum Iterations", 200)),
-    m_myPID(Preconditioner->MyPID()),
-    m_printFlag(m_Parameters->get("Print Summary", 0)),
-    m_numIterTotal(0),
-    m_numSolvesTotal(0),
-    m_convergenceCheckInterval
-    (Parameters->get("Convergence Check Interval", 10)),
-    m_krylovMethod(Parameters->get("Krylov Method", 0)),
-    m_estimateConditionNumber(Parameters->get("Estimate Condition Number", 0)),
-    m_solverTol(Parameters->get("Convergence Tolerance", 1e-6)),
-    m_totalTime(0),
-    m_relativeResidual(0),
-    m_relativeResidualActual(0),
-    m_originalNorm(0),
-    m_currentNorm(0),
-    m_Orthog(Teuchos::null)
+    m_Parameters(Parameters)
   {
-    if (m_printFlag > 1) m_calculateRelativeResidualActual = true;
-    if (m_myPID != 0) m_printFlag = 0;
-    reserveMemory();
-    if (m_krylovMethod == GCR) {
-      m_Orthog = Teuchos::rcp( new OrthogGCR<SX,SM,LO,GO>
-			       (m_Preconditioner,
-				m_Parameters,
-				m_krylovLength) );
-    }
+    initialize();
   }
    
   ~KrylovSolver()
```

Comment:
`~KrylovSolver() = default;`

Reply:
Done.

Reply:
We had a chat about this over the phone today.